pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -296,6 +297,37 @@ func createTargetConfigController(ctx context.Context, syncCtx factory.SyncConte
 		return true, nil
 	}
 
+	// Set CloudControllerOwner condition to False if external cloud provider has been successfully applied.
+	externalCloudProviderApplied, err := isExternalCloudProviderApplied(operatorSpec)
+	if err != nil {
+		return true, err
+	}
+	if externalCloudProviderApplied {
+		_, status, _, err := c.operatorClient.GetStaticPodOperatorState()
+		if err != nil {
+			return true, nil
+		}
+
+		// Once we have successfully set CloudControllerOwner to False, we should not change it again.
+		if !v1helpers.IsOperatorConditionFalse(status.Conditions, "CloudControllerOwner") {
+			cond := operatorv1.OperatorCondition{
+				Type:   "CloudControllerOwner",
+				Status: operatorv1.ConditionTrue,
+			}
+
+			// We set the condition to False only if at least one static pod has successfully started. After that
+			// we are sure that KCM doesn't own CloudController anymore. Otherwise we set it to True by implying
+			// that KCM is still the owner.
```

Comment:
Why does one static pod starting mean that KCM doesn't own it anymore? Should we not wait for them all to be updated to the new version?

Reply:
Now we check `NodeInstallerProgressingCondition` condition to make sure that all pods are at the latest revision